application/node.go

```diff
@@ -273,7 +300,15 @@ func (n *Node) emitNodeEvent(event cc.Command) {
 
 func (n *Node) receiveControllerUpdate(update serialapi.ControllerUpdate) {
 	n.setFromApplicationControllerUpdate(update)
+
+	select {
+	case n.queryStageNIFComplete <- true:
+	default:
+	}
+
+	n.QueryStageNIF = true
 	n.saveToDb()
+	n.nextQueryStage()
```

Comment:
Nothing jumps out at me here code-wise. Not sure what this does well enough :). @bjyoungblood might have a better idea.